### PR TITLE
Add Quality Declaration level 4 ament_index_cpp

### PR DESCRIPTION
**Pull request by Blast545**
This is a quality declaration for ament_index_cpp as described in the proposed REP-2004 (see ros-infrastructure/rep#218).

Signed-off-by: Jorge Perez jjperez@ekumenlabs.com

### CODE REVIEW

File: ament_index_cpp/Quality_Declaration.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
+This document is a declaration of software quality for the `ament_index_cpp` package, based on the guidelines in [REP-2004](https://github.com/ros-infrastructure/rep/blob/rep-2004/rep-2004.rst).
```
Comment (brawner) [inline ament_index_cpp/Quality_Declaration.md]: Rename to QUALITY_DECLARATION.md

**Review by brawner — APPROVED** [review]

File: ament_index_cpp/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# ament_index_cpp
+
+`ament_index_cpp` is a C++ API to access the ament resource index.
+
```
Comment (ahcorde) [inline ament_index_cpp/README.md:4]: we should include the features in the README.md or Doxygen

**Review by ahcorde — CHANGES_REQUESTED** [review]
Please add the features of this package

File: ament_index_cpp/README.md (hunk 2 above)
Comment (Blast545) [reply ament_index_cpp/README.md:4]: I just assigned myself to that. However, it is not directly related to this PR, shouldn't stop merging this

**Pull request merged**